lib_standard_app/main.c

```diff
@@ -80,6 +89,76 @@ WEAK __attribute__((section(".boot"))) int main() {
     }
 
     app_exit();
+}
+
+#ifdef HAVE_SWAP
+static void library_app_main_helper(struct libargs_s *args) {
+    PRINTF("Inside library\n");
+    switch (args->command) {
+        case SIGN_TRANSACTION:
+            if (swap_copy_transaction_parameters(args->create_transaction)) {
+                common_app_init();
+
+                G_called_from_swap = true;
+                G_swap_response_ready = false;
+
+                app_main();
+            }
+            break;
+        case CHECK_ADDRESS:
+            // ensure result is zero if an exception is thrown
+            args->check_address->result = 0;
```

Comment:
Not necessary, already done in app-exchange before the call.

Reply:
Not a bad idea to avoid assuming things from an app in the sdk